examples/direct_fidelity_estimation.py

```diff
@@ -57,25 +56,31 @@ def main():
   # Computes for every \hat{P_i} of https://arxiv.org/pdf/1104.3835.pdf,
   # estimate rho_i and Pr(i). We then collect tuples (rho_i, Pr(i), \hat{Pi})
   # inside the variable 'pauli_traces'.
-  pauli_traces = [
-      compute_characteristic_function(circuit, P_i, qubits, noise=None) + (P_i,)
-      for P_i in itertools.product([cirq.I, cirq.X, cirq.Y, cirq.Z],
-                                           repeat=n)
-  ]
-  highest_probs = heapq.nlargest(n, pauli_traces, key=lambda e: e[1])
+  pauli_traces = []
+  for P_i in itertools.product([cirq.I, cirq.X, cirq.Y, cirq.Z], repeat=n):
+    rho_i, Pr_i = compute_characteristic_function(
+        circuit, P_i, qubits, noise=None)
+    pauli_traces.append({'P_i': P_i, 'rho_i': rho_i, 'Pr_i': Pr_i})
+
+  assert len(pauli_traces) == 4**n
+
+  p = [x['Pr_i'] for x in pauli_traces]
+  assert numpy.isclose(sum(p), 1.0, 1e-6)
```

Comment:
nit: use a kwarg to document what 1e-6 is. Namely: `atol` or `rtol`. I legitimately don't know which one it actually is since I don't remember the function signature order :)

Reply:
It was 'rtol' but I think actually that 'atol' makes more sense, and so I used this.